crates/uv/src/commands/project/lock.rs

```diff
@@ -334,6 +334,17 @@ impl<'env> LockOperation<'env> {
                     .read()
                     .await?
                     .ok_or_else(|| ProjectError::MissingLockfile)?;
+                // Check if the discovered workspace members match the locked workspace members.
+                if let LockTarget::Workspace(workspace) = target {
+                    for package_name in workspace.packages().keys() {
+                        existing
+                            .find_by_name(package_name)
+                            .map_err(|_| ProjectError::LockWorkspaceMismatch(package_name.clone()))?
+                            .ok_or_else(|| {
+                                ProjectError::LockWorkspaceMismatch(package_name.clone())
+                            })?;
+                    }
+                }
```

Comment:
This is technically more strict than the previous check, but it will only fail in cases where the lockfile doesn't match the on-disk workspace structure, while afaik we require the workspace structure to be intact through `VirtualProject::discover` (`uv export --frozen --no-emit-workspace --no-emit-project` panics too)?